### PR TITLE
[OF-1438] Templated Json Serializer

### PR DESCRIPTION
**Pull request by MAG-mv**
- Added support for templated serialization/deserialization to easily serialize objects without modifying existing json code
- Added tests for current features.

### CODE REVIEW

File: Library/src/Json/JsonSerializer.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 Magnopus LLC
```
Comment (MAG-OR) [inline Library/src/Json/JsonSerializer.cpp:2]: I was asking myself the question because I have a new file as well - are we supposed to put 2024 there?

**Review by MAG-ChristopherAtkinson — APPROVED** [review]
The changes look great, added some minor comments but don't effect the overall behaviour, happy to approve.

File: Library/src/Json/JsonSerializer.cpp (hunk 1 above)
Comment (MAG-mv) [reply Library/src/Json/JsonSerializer.cpp:2]: Good question. I will find out

Comment (MAG-mv) [reply Library/src/Json/JsonSerializer.cpp:2]: @MAG-AdamThorn Do you have any information on this?

File: Tests/src/InternalTests/JsonTests.cpp [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+{
+	int32_t Int32Member				 = 0;
+	uint32_t Uint32Member			 = 0;
+	int64_t Int64Member				 = 0;
+	uint64_t Uint64Member			 = 0;
+	float FloatMember				 = 0.f;
+	float DoubleMember				 = 0.0;
+	csp::common::String StringMember = "";
+	const char* CharPtrMember		 = "";
+};
```
[the remark this answers is not in the document]
Comment (MAG-mv) [reply Tests/src/InternalTests/JsonTests.cpp:35]: This looks correct on my end. I'm not sure why Github is displaying it like this

File: Library/src/Json/JsonSerializer.h [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+		Serializer.SerializeValue(Object);
+		return Serializer.Buffer.GetString();
+	}
+
+	/// @brief Should be called within custon ToJson function
+	/// This will serialize a member
+	/// If the member is another custom type, this was internally call ToJson on that type
+	/// @param Key const char * Key : The key to reference the item
+	/// @param Value const T& : The value to serialize
+	template <typename Val> void SerializeMember(const char* Key, const Val& Value)
```
[the remark this answers is not in the document]
Comment (MAG-mv) [reply Library/src/Json/JsonSerializer.h]: Updated

**Review by MAG-AdamThorn — APPROVED** [review]

**Review by MAG-ChristopherAtkinson — APPROVED** [review]

**Pull request merged**